### PR TITLE
Refactor PlatformComponent. Remove platform-specific classes

### PR DESCRIPTION
**Pull request by igordmn**
The idea is to abstract away this interface from any platform specific classes. If this interface don't have these classes (`actual`'s as well), it can be used for different implementations on single target. For example for JavaFX, GLFW, etc on java target.

Also refactor:
- pointer icon
- accessibility
- input methods

### CODE REVIEW

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/Platform.skiko.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import androidx.compose.ui.input.pointer.PointerIcon
+import androidx.compose.ui.node.LayoutNode
+import androidx.compose.ui.semantics.SemanticsOwner
+import androidx.compose.ui.text.input.EditCommand
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.ImeOptions
+import androidx.compose.ui.text.input.PlatformTextInputService
+import androidx.compose.ui.text.input.TextFieldValue
+
+// TODO(demin): make it public when we stabilize it after implementing it for uikit and js
```
Comment (dima-avdeev-jb) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/Platform.skiko.kt:28]: will there be a compilation error for uikit and js now?

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt [hunk 2]
```diff
@@ -91,9 +95,24 @@ internal class ComposeLayer {
         }
     }
 
+    private val platform = object : Platform {
+        override fun setPointerIcon(pointerIcon: PointerIcon) {
+            if (pointerIcon is AwtCursor) {
+                _component.cursor = pointerIcon.cursor
+            }
```
Comment (dima-avdeev-jb) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt]: maybe we need else branch with default cursor?

**Review by dima-avdeev-jb — APPROVED** [review]

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/Platform.skiko.kt (hunk 1 above)
Comment (igordmn) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/Platform.skiko.kt:28]: Actually, there was compilation error 😅. Fixed

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt (hunk 2 above)
Comment (igordmn) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt]: Good idea. Fixed

**Pull request merged**